src/octoprint/static/js/app/viewmodels/terminal.js

```diff
@@ -259,8 +259,12 @@ $(function() {
 
             var re = /^([gmt][0-9]+)(\s.*)?/;
             var commandMatch = command.match(re);
+            self.blacklist =  splitTextToArray(self.settings.serial_autoUppercaseBlacklist(), ",", true);
```

Comment:
Setting an (undeclared) instance variable like this within a method is unclean. Especially since you don't use this instance variable outside of this method.

Two ways to go about this. One is simply declaring a local `var blacklist` instead of `self.blacklist`. That has the disadvantage however that on ever single command that is send the whole blacklist will be fetched from the settings view model and turned into an array before matching.

Cleaner would be to use `self.settings.serial_autoUppercaseBlacklist.subscribe(...)` to be notified of changes to the value (see [Kockout's docs for details on subscribe](http://knockoutjs.com/documentation/observables.html#explicitly-subscribing-to-observables)) and then updating a (previously declared) `self.blacklist` with the array-ified value that in turn is then used in the `sendCommand` routine.

So basically: 

  * add `self.blacklist` to the view model properly (top area), initialize to empty array
  * subscribe to `self.settings.serial_autoUppercaseBlacklist`:

    ``` javascript
    self.settings.serial_autoUppercaseBlacklist(function(newValue) {
        self.blacklist = splitTextToArray(newValue, ",", true);
    });
    ```
  * Use `self.blacklist` in `sendCommand`